### PR TITLE
Updates as per Laravel 7 upgrade guide

### PR DESCRIPTION
**Pull request by edalzell**
Not sure if this is something you're open to, but this seems to get Statamic up to Laravel 7, although you need to wait for https://github.com/mrclay/minify/pull/666 to actually get there.

However all of these changes should work for L5.8+ regardless to be ready!

### CODE REVIEW

File: src/Http/Resources/API/AssetResource.php [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 namespace Statamic\Http\Resources\API;
 
-use Illuminate\Http\Resources\Json\Resource;
+use Illuminate\Http\Resources\Json\JSONResource;
 
-class AssetResource extends Resource
+class AssetResource extends JSONResource
```
Comment (jasonvarga) [inline src/Http/Resources/API/AssetResource.php]: These should extend JsonResource. The case needs to be adjusted.

**Comment by jasonvarga** [comment]
Changing the exception handler class will work for Laravel 7 but break everything else. I think we need to work out how to get rid of it. Related to #1295

**Comment by edalzell** [comment]
New PR in `minify` to watch for: https://github.com/mrclay/minify/pull/667

**Comment by riasvdv** [comment]
Seems like the minify PR Is merged!

**Comment by edalzell** [comment]
> Changing the exception handler class will work for Laravel 7 but break everything else. I think we need to work out how to get rid of it. Related to #1295

@jasonvarga do think this is something one of us could tackle or will this require a deeper knowledge of how Statamic is all wired up? If the former, do you have a suggested starting point?

File: src/Providers/ExtensionServiceProvider.php [hunk 2]
```diff
@@ -6,19 +6,14 @@
...
 use Illuminate\Filesystem\Filesystem;
-use Illuminate\Support\Facades\Cache;
+use Statamic\Modifiers\CoreModifiers;
 use Illuminate\Support\ServiceProvider;
-use Statamic\Extend\Manifest;
-use Illuminate\Console\DetectsApplicationNamespace;
 
 class ExtensionServiceProvider extends ServiceProvider
 {
-    use DetectsApplicationNamespace;
```
Comment (jasonvarga) [inline src/Providers/ExtensionServiceProvider.php:20]: Why did you remove this?

Comment (jasonvarga) [reply src/Providers/ExtensionServiceProvider.php:20]: Nevermind, I see it was deprecated.

**Comment by edalzell** [comment]
Official PR https://github.com/statamic/cms/pull/1679

**Comment by jasonvarga** [comment]
Leave it open and you can get a cool purple merged badge.
